## 1. Importing the library

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
print('Libraries imported sucessfully')

Libraries imported sucessfully


In [2]:
import warnings
warnings.filterwarnings('ignore')

## 2. Loading the dataset

In [4]:
sns.get_dataset_names

<function seaborn.utils.get_dataset_names()>

In [5]:
df = sns.load_dataset('iris')

In [6]:
# first 5 records
df.head()

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

In [7]:
# Last 5 records
df.tail()

sepal_length  sepal_width  petal_length  petal_width    species
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

In [8]:
# Shape
df.shape

(150, 5)

In [9]:
print('The number of rows are :',df.shape[0])
print('The number of columns are :',df.shape[1])

The number of rows are : 150
The number of columns are : 5


In [10]:
# File summary of the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In this dataset there are total 150 recoreds are present and with 4 columns.

4 features are numerical feature and 1 feature is categorical feature.

In [11]:
# Check description of dataset
df.describe()

sepal_length  sepal_width  petal_length  petal_width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.057333      3.758000     1.199333
std        0.828066     0.435866      1.765298     0.762238
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000

In [12]:
# Check for missing value
df.isna().sum()

sepal_length    0
sepal_width     0
petal_length    0
petal_width     0
species         0
dtype: int64

There are no missing values are present in the dataset. 

In [14]:
# Check for duplicate
df.duplicated().sum()

1

1 duplicate row is found in the dataset.

In [17]:
# Now remove the duplicate rows
df.drop_duplicates(inplace=True)

In [18]:
df.duplicated().sum()

0

So now no duplicates row is present then we can proceed further.

In [19]:
# How to select nan values

In [20]:
df[df.isna().any(axis=1)]

Empty DataFrame
Columns: [sepal_length, sepal_width, petal_length, petal_width, species]
Index: []

In [21]:
df[:5]

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

In [22]:
# drop sepal_length column
df.drop(labels='sepal_length',axis=1)

sepal_width  petal_length  petal_width    species
0            3.5           1.4          0.2     setosa
1            3.0           1.4          0.2     setosa
2            3.2           1.3          0.2     setosa
3            3.1           1.5          0.2     setosa
4            3.6           1.4          0.2     setosa
..           ...           ...          ...        ...
145          3.0           5.2          2.3  virginica
146          2.5           5.0          1.9  virginica
147          3.0           5.2          2.0  virginica
148          3.4           5.4          2.3  virginica
149          3.0           5.1          1.8  virginica

[149 rows x 4 columns]

In [23]:
# How to drop misssing values

In [24]:
df.dropna()

sepal_length  sepal_width  petal_length  petal_width    species
0             5.1          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[149 rows x 5 columns]

In [25]:
# list the categories in the species
df.species.unique()

array(['setosa', 'versicolor', 'virginica'], dtype=object)

In [27]:
df.species.value_counts()

setosa        50
versicolor    50
virginica     49
Name: species, dtype: int64

In [32]:
a = df.groupby(['species']).mean()

In [33]:
a

sepal_length  sepal_width  petal_length  petal_width
species                                                         
setosa          5.006000     3.428000      1.462000     0.246000
versicolor      5.936000     2.770000      4.260000     1.326000
virginica       6.604082     2.979592      5.561224     2.028571

In [34]:
df.groupby(['species']).max()

sepal_length  sepal_width  petal_length  petal_width
species                                                         
setosa               5.8          4.4           1.9          0.6
versicolor           7.0          3.4           5.1          1.8
virginica            7.9          3.8           6.9          2.5

In [35]:
df.groupby(['species']).min()

sepal_length  sepal_width  petal_length  petal_width
species                                                         
setosa               4.3          2.3           1.0          0.1
versicolor           4.9          2.0           3.0          1.0
virginica            4.9          2.2           4.5          1.4